hub/core/storage/tests/test_storage_provider.py

```diff
@@ -125,11 +126,12 @@ def test_cache(storage):
 
 @parametrize_all_storages
 def test_pickling(storage):
-    FILE_1 = f"{KEY}_1"
-    storage[FILE_1] = b"hello world"
-    assert storage[FILE_1] == b"hello world"
-    pickle_file = open("storage_pickle", "wb")
-    pickle.dump(storage, pickle_file)
-    pickle_file = open("storage_pickle", "rb")
-    unpickled_storage = pickle.load(pickle_file)
-    assert unpickled_storage[FILE_1] == b"hello world"
+    with CliRunner().isolated_filesystem():
+        FILE_1 = f"{KEY}_1"
+        storage[FILE_1] = b"hello world"
+        assert storage[FILE_1] == b"hello world"
+        pickle_file = open("storage_pickle", "wb")
+        pickle.dump(storage, pickle_file)
+        pickle_file = open("storage_pickle", "rb")
+        unpickled_storage = pickle.load(pickle_file)
+        assert unpickled_storage[FILE_1] == b"hello world"
```

Comment:
what is this change? is it relevant to the messages?

Reply:
Not really. This is related to a temp file being created while running `pytest`. This code ensures that the temp files created by this function will be deleted, even if the test fails.